plutus-benchmark/validation/Common.hs

```diff
@@ -120,16 +123,28 @@ parserInfo cfg =
    or
      `cabal bench -- plutus-benchmark:validation --benchmark-options crowdfunding`.
 -}
-main :: IO ()
-main = do
-  cfg <- getConfig 20.0  -- Run each benchmark for at least 20 seconds.  Change this with -L or --timeout (longer is better).
-  options <- execParser $ parserInfo cfg
-  scriptDirectory <- getScriptDirectory
-  files0 <- listDirectory scriptDirectory  -- Just the filenames, not the full paths
-  let files1 = naturalSort $ filter (isExtensionOf ".flat") files0  -- Just in case there's anything else in the directory.
-               -- naturalSort puts the filenames in a better order than Data.List.Sort
-      files = if quick options then files1 `withAnyPrefixFrom` quickPrefixes else files1
-      benchmarks = mkBMs scriptDirectory files
-  runMode (otherOptions options) benchmarks
+validationBench :: ValidationMode -> IO ()
+validationBench vMode = do
+    cfg <- getConfig 20.0  -- Run each benchmark for at least 20 seconds.  Change this with -L or --timeout (longer is better).
+    options <- execParser $ parserInfo cfg
+    scriptDirectory <- getScriptDirectory
+    files0 <- listDirectory scriptDirectory  -- Just the filenames, not the full paths
+    let files = if quick options then files1 `withAnyPrefixFrom` quickPrefixes else files1
+        -- naturalSort puts the filenames in a better order than Data.List.Sort
+        files1 = naturalSort $ filter (isExtensionOf ".flat") files0  -- Just in case there's anything else in the directory.
+    runMode (otherOptions options) $ mkBMs scriptDirectory files
+ where
+
+    -- Make benchmarks for the given files in the directory
+    mkBMs :: FilePath -> [FilePath] -> [Benchmark]
+    mkBMs dir files = map (mkScriptBM dir) files
+
+    mkScriptBM :: FilePath -> FilePath -> Benchmark
+    mkScriptBM dir file =
+        env (readFlat $ dir </> file) $ bench (dropExtension file) .
```

Comment:
This took me a while to parse. I think it's a bit weird composing with a case expression, can you just eta-expand this so it's more obvious where the env is coming in?